api/v1alpha2/azuremachine_conversion.go

```diff
@@ -67,6 +67,9 @@ func restoreAzureMachineSpec(restored, dst *infrav1alpha3.AzureMachineSpec) {
 	}
 	dst.OSDisk.DiffDiskSettings = restored.OSDisk.DiffDiskSettings
 	dst.OSDisk.CachingType = restored.OSDisk.CachingType
+	if restored.OSDisk.ManagedDisk.DiskEncryptionSet != nil {
+		dst.OSDisk.ManagedDisk.DiskEncryptionSet = restored.OSDisk.ManagedDisk.DiskEncryptionSet.DeepCopy()
```

Comment:
try running `make generate`, I see `generated files are out of date, run make generate` in CI